File: 0896-IsMonotonic/0896.java
class Solution {
  public boolean isMonotonic(int[] A) {
    boolean inc = true, dec = true;
    int n = A.length;
    for (int i = 0; i < n - 1; ++i) {
      if (A[i] > A[i + 1]) {
        inc = false;
      }
      if (A[i] < A[i + 1]) {
        dec = false;
      }
    }
  }
  return inc || dec;
}

File: 801-900/0896-IsMonotonic/0896.py
class Solution:
  def isMonotonic(self, A):
    N = len(A)
    inc, dec = True, True
    for i in range(1, N):
      if A[i] > A[i + 1]:
        inc = False
      if A[i] < A[i + 1]:
        dec = False
      if not inc and not dec:
        return False
    return True